views/Products/Create/Form.tsx

```diff
@@ -1,16 +1,41 @@
-import { FC } from 'react'
-import { useForm } from 'react-hook-form'
+import Image from 'next/image'
+import { ChangeEvent, FC, useState } from 'react'
+import { SubmitHandler, useForm } from 'react-hook-form'
+import Swal from 'sweetalert2'
 
 import { MutableProductFormProps, ProductEditable } from '@/services/GraphQL/products/types'
+import stylesInput from '@/styles/EditEvent.module.scss'
+
+interface ProductEditableWithImg extends ProductEditable {
+  image?: FileList
+}
 
 const CreateProductForm: FC<MutableProductFormProps> = ({ onSubmit, loading }) => {
+  const [imageFile, setImageFile] = useState<File | null>(null)
+  const [imageUrl, setImageUrl] = useState<string | null>(null)
   const {
     register,
     handleSubmit,
     reset,
     formState: { errors },
-  } = useForm<ProductEditable>()
-  const submitHandler = handleSubmit(onSubmit)
+  } = useForm<ProductEditableWithImg>()
+  const submitHandler = handleSubmit(onSubmit as unknown as SubmitHandler<ProductEditableWithImg>)
+  const handleFile = (evt: ChangeEvent<HTMLInputElement>) => {
+    const file = evt.target.files?.[0]
+    if (file && file.size >= 8000000) {
```

Comment:
Please avoid _magic numbers_ (define number in constant)
```suggestion
    if (file && file.size >= MAX_FILE_SIZE) {
```